### PR TITLE
STOR-1312: Add openshift/secrets-store-csi-driver repo

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/STOR-1312
/cc @openshift/storage @jsafrane @gnufied

### CODE REVIEW

**Comment by dobsonj** [comment]
/assign @dhellmann
for image mirroring approval

File: ci-operator/config/openshift/secrets-store-csi-driver/openshift-secrets-store-csi-driver-main.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    name: "4.14"
+    namespace: ocp
+    tag: base
+  ocp_builder_rhel-8-golang-1.20-openshift-4.14:
+    name: builder
+    namespace: ocp
+    tag: rhel-8-golang-1.20-openshift-4.14
+binary_build_commands: make build
+build_root:
+  from_repository: true
```
Comment (mpatlasov) [inline ci-operator/config/openshift/secrets-store-csi-driver/openshift-secrets-store-csi-driver-main.yaml:12]: Why not add the line `canonical_go_repository: sigs.k8s.io/secrets-store-csi-driver` after this line? (see this place in a similar file https://github.com/openshift/release/blob/master/ci-operator/config/openshift/ibm-vpc-block-csi-driver/openshift-ibm-vpc-block-csi-driver-master.yaml#L13)

**Comment by mpatlasov** [comment]
The PR would be much simpler to review if it was clear how questions from `make new-repo` were answered, and which parts of auto-generated `ci-operator/config/openshift/secrets-store-csi-driver/openshift-secrets-store-csi-driver-main.yaml`  were manually modified and why. Having said that, the diff with similar file in `ibm-vpc-block-csi-driver` didn't show any weird discrepancies. Only `canonical_go_repository: ...` is unclear.

File: ci-operator/config/openshift/secrets-store-csi-driver/openshift-secrets-store-csi-driver-main.yaml (hunk 1 above)
Comment (dobsonj) [reply ci-operator/config/openshift/secrets-store-csi-driver/openshift-secrets-store-csi-driver-main.yaml:12]: Thanks, I added it in a new commit. I actually wasn't certain about this line, as we have a few drivers like google filestore without it, but it appears to be [purely cosmetic](https://github.com/openshift/ci-tools/blob/a8bae5fb2eaec077982e90430dde7407d5d82fd2/pkg/api/types.go#L58-L62).

**Comment by dobsonj** [comment]
> The PR would be much simpler to review if it was clear how questions from `make new-repo` were answered, and which parts of auto-generated `ci-operator/config/openshift/secrets-store-csi-driver/openshift-secrets-store-csi-driver-main.yaml` were manually modified and why. Having said that, the diff with similar file in `ibm-vpc-block-csi-driver` didn't show any weird discrepancies. Only `canonical_go_repository: ...` is unclear.

I'll include the answers I provided to `make new-repo` here for future reference, but I must have answered part of it wrong, so it might not be a good example. I had to add the `tests:` and `images:` sections by hand.

```
$ make new-repo
false || docker pull --platform linux/amd64 registry.ci.openshift.org/ci/repo-init:latest
latest: Pulling from ci/repo-init
Digest: sha256:13394c7db21125fe86a7c3710761b41f22529f43eaca30fb6b0ba6f31cb02a85
Status: Image is up to date for registry.ci.openshift.org/ci/repo-init:latest
registry.ci.openshift.org/ci/repo-init:latest
docker run --user 1000:1000 --platform linux/amd64 --rm -it -v "/home/jon/workspace/release:/release:z" registry.ci.openshift.org/ci/repo-init:latest --release-repo /release
Welcome to the repository configuration initializer.
In order to generate a new set of configurations, some information will be necessary.

Let's start with general information about the repository...
Enter the organization for the repository: openshift
Enter the repository to initialize: secrets-store-csi-driver
Enter the development branch for the repository: [default: master] main

Now, let's determine how the repository builds output artifacts...
Does the repository build and promote container images?  [default: no] yes
Does the repository promote images as part of the OpenShift release?  [default: no]
Do any images build on top of the OpenShift base image?  [default: no] yes
Do any images build on top of the CentOS base image?  [default: no]

Now, let's configure how the repository is compiled...
What version of Go does the repository build with? [default: 1.13] 1.20
[OPTIONAL] Enter the Go import path for the repository if it uses a vanity URL (e.g. "k8s.io/my-repo"):
[OPTIONAL] What commands are used to build binaries in the repository? (e.g. "go install ./cmd/...") make build
[OPTIONAL] What commands are used to build test binaries? (e.g. "go install -race ./cmd/..." or "go test -c ./test/...") make test

Now, let's configure test jobs for the repository...

First, we will configure simple test scripts. Test scripts
execute unit or integration style tests by running a command
from your repository inside of a test container. For example,
a unit test may be executed by running "make test-unit" after
checking out the code under test.

Are there any test scripts to configure?  [default: no]

Next, we will configure end-to-end tests. An end-to-end test
executes a command from your repository against an ephemeral
OpenShift cluster. The test script will have "cluster:admin"
credentials with which it can execute no other tests will
share the cluster.

Are there any end-to-end test scripts to configure?  [default: no]

Repository configuration options loaded!
In case of any errors, use the following command to re-
create this run without using the interactive interface:

/usr/bin/repo-init --release-repo /release --config="{\"org\":\"openshift\",\"repo\":\"secrets-store-csi-driver\",\"branch\":\"main\",\"base_images\":null,\"images\":null,\"canonical_go_repository\":\"\",\"promotes\":true,\"promotes_with_openshift\":false,\"needs_base\":true,\"needs_os\":false,\"go_version\":\"1.20\",\"build_commands\":\"make build\",\"test_build_commands\":\"make test\",\"tests\":null,\"custom_e2e\":null,\"release_type\":\"\",\"release_version\":\"\",\"operator_bundle\":null}"

Updating Prow configuration ...

Updating Prow plugin configuration ...
INFO[0241] Generating CI Operator configuration ...
make update
make[1]: Entering directory '/home/jon/workspace/release'
make jobs
make[2]: Entering directory '/home/jon/workspace/release'
false || docker pull --platform linux/amd64 registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
latest: Pulling from ci/ci-operator-checkconfig
ed7494878b3d: Already exists
58b4858ef9a2: Pull complete
Digest: sha256:2eb917f06c81730b419731b9cd481b32b00979a72a2eb25e6febadd6742884ed
Status: Downloaded newer image for registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
docker run --user 1000:1000 --platform linux/amd64 --rm -v "/home/jon/workspace/release/ci-operator/config:/ci-operator/config:z" -v "/home/jon/workspace/release/ci-operator/step-registry:/ci-operator/step-registry:z" registry.ci.openshift.org/ci/ci-operator-checkconfig:latest --config-dir /ci-operator/config --registry /ci-operator/step-registry
time="2023-05-22T20:57:13Z" level=error msg="Failed to load CI Operator configuration" error="invalid ci-operator config: invalid configuration: you must define at least one test or image build in 'tests' or 'images'" source-file=/ci-operator/config/openshift/secrets-store-csi-driver/openshift-secrets-store-csi-driver-main.yaml
time="2023-05-22T20:57:14Z" level=error error="error reading configuration files: invalid ci-operator config: invalid configuration: you must define at least one test or image build in 'tests' or 'images'"  
time="2023-05-22T20:57:14Z" level=fatal msg="error validating configuration files"
make[2]: *** [Makefile:77: ci-operator-checkconfig] Error 1
make[2]: Leaving directory '/home/jon/workspace/release'
make[1]: *** [Makefile:55: update] Error 2
make[1]: Leaving directory '/home/jon/workspace/release'
make: *** [Makefile:111: new-repo] Error 2
```

**Comment by dobsonj** [comment]
Ok I pushed the following commits and pj-rehearse is finally happy:

add canonical_go_repository: https://github.com/openshift/release/pull/39649/commits/938eb6dd50082aefeb3ca3391786ecbc659d9855
add GOFLAGS="-mod=readonly": https://github.com/openshift/release/pull/39649/commits/fef9b02e5b02606b661cc5e520061dcce86213b6
add GOCACHE=/tmp and XDG_CACHE_HOME=/tmp: https://github.com/openshift/release/pull/39649/commits/10cc24a06c594f6399dbf0710b86ce02382a35e4

I just squashed those into a single commit, so one more run and hopefully it will be mergeable.
/pj-rehearse

**Comment by dobsonj** [comment]
> @dobsonj: all tests **passed!**

/pj-rehearse ack
/cc @jsafrane @mpatlasov 
(for re-lgtm)

**Review by dhellmann — COMMENTED** [review]
/approve

**Pull request merged**